lib/secure_headers.rb

```diff
@@ -45,181 +100,158 @@ def configure &block
   class << self
     def append_features(base)
       base.module_eval do
-        extend ClassMethods
         include InstanceMethods
       end
     end
 
-    def header_hash(options = nil)
+    # Strips out headers not applicable to this request
+    def header_hash_for(request)
+      unless request.scheme == HTTPS
+        secure_headers_request_config(request)[SecureHeaders::StrictTransportSecurity::CONFIG_KEY] = SecureHeaders::OPT_OUT
+        secure_headers_request_config(request)[SecureHeaders::PublicKeyPins::CONFIG_KEY] = SecureHeaders::OPT_OUT
+      end
+
       ALL_HEADER_CLASSES.inject({}) do |memo, klass|
-        config = if options.is_a?(Hash) && options[klass::Constants::CONFIG_KEY]
-          options[klass::Constants::CONFIG_KEY]
+        header_config = request.env[klass::CONFIG_KEY] ||
+          secure_headers_request_config(request)[klass::CONFIG_KEY]
+
+        header_name, header = if header_config
+          make_header(klass, header_config, request.user_agent)
         else
-          ::SecureHeaders::Configuration.send(klass::Constants::CONFIG_KEY)
+          # use the cached default, if available
+          if default_header = SecureHeaders::Configuration::default_headers[klass::CONFIG_KEY]
```

Comment:
This is a weird syntax for accessing a method. How about `SecureHeaders::Configuration.default_headers`, unless you think `::` is more idiomatic.